ui/src/app/analytics/analytics.service.ts

```diff
@@ -1,39 +1,115 @@
 import { Injectable } from '@angular/core';
-import { Http, Response } from '@angular/http';
+import { Http, Response, RequestOptionsArgs } from '@angular/http';
+
 import { Observable } from 'rxjs/Observable';
+import { Subscription } from 'rxjs/Subscription';
 import 'rxjs/add/operator/catch';
 import 'rxjs/add/operator/map';
 
+import { ErrorHandler, Page } from '../shared/model';
+import { Counter } from './counters/model/counter.model';
+import { HttpUtils } from '../shared/support/http.utils';
+
+/**
+ * @author Gunnar Hillert
+ */
 @Injectable()
 export class AnalyticsService {
 
-  private appstUrl = '/apps';
+  private metricsCountersUrl = '/metrics/counters';
+
+  public counters: Page<Counter>;
+  public _counterInterval = 2;
+  public counterPoller: Subscription;
+  public countersCache: number[];
 
-  constructor(private http: Http) { }
+  constructor(private http: Http, private errorHandler: ErrorHandler) {
+  }
 
-  getApps(): Observable<any[]> {
-    return this.http.get(this.appstUrl)
-                    .map(this.extractData)
-                    .catch(this.handleError);
+  set counterIntervall(rate: number) {
+    if (rate && !isNaN(rate)) {
+      if (rate < 0.01) {
+          rate = 0;
+          this.stopPollingForCounters();
+      } else {
+        console.log('Setting intervall to ' + rate);
+        this._counterInterval = rate;
+        if (this.counterPoller && !this.counterPoller.closed) {
+          this.stopPollingForCounters();
+          this.startPollingForCounters();
+        }
+      }
+    }
   }
 
-  private extractData(res: Response) {
-    const body = res.json();
-    return body._embedded;
+  get counterIntervall(): number {
+    return this._counterInterval;
   }
 
-  private handleError (error: Response | any) {
-    // In a real world app, you might use a remote logging infrastructure
-    let errMsg: string;
-    if (error instanceof Response) {
-      const body = error.json() || '';
-      const err = body.error || JSON.stringify(body);
-      errMsg = `${error.status} - ${error.statusText || ''} ${err}`;
-    } else {
-      errMsg = error.message ? error.message : error.toString();
+  public totalCacheSize() {
+    return Math.max(Math.ceil(60 / this._counterInterval), 20);
+  }
+
+  startPollingForCounters() {
+    if (!this.counterPoller || this.counterPoller.closed) {
+      this.counterPoller = Observable.interval(this._counterInterval * 1000)
+        .switchMap(() => this.getAllCounters(true)).subscribe();
     }
-    console.error(errMsg);
-    return Observable.throw(errMsg);
   }
 
+  stopPollingForCounters() {
```

Comment:
I don't see this called from anywhere if you navigate away from analytics. Thus unneeded requests are made unless you actually go back to counters tab and set interval to 0. Setting interval to 0 stop poller but then it's not reactivated if you set interval to 1(thus you need to go some other tab and go back to counters tab which activates poller again)